### DISCUSSION TITLE
How to use Yazi as a file picker for Firefox?

### DISCUSSION THREAD

**Discussion by xronin01** [sxyazi/yazi#1875] [1. Q&A]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

wezterm

### `yazi --debug` output

```Shell
.
```

### Describe the question

.

### Validations

- [X] I have searched the existing discussions/issues

**Comment by boydaihungst** [comment]
https://github.com/boydaihungst/xdg-desktop-portal-termfilechooser

**Reply by boydaihungst** [reply]
Remember to set `TERMCMD` with your terminal open command. By default, it uses `kitty`. In your case this should works: `wezterm start --always-new-process`

**Reply by xronin01** [reply]
compilation error in debian

ninja: Entering directory `build'
[1/2] Compiling C object xdg-desktop-portal-termfilechooser.p/src_filechooser_filechooser.c.o
FAILED: xdg-desktop-portal-termfilechooser.p/src_filechooser_filechooser.c.o 
cc -Ixdg-desktop-portal-termfilechooser.p -I. -I.. -I../include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O0 -g -Wno-missing-braces -Wno-missing-field-initializers -Wno-unused-parameter -D_POSIX_C_SOURCE=200809L '-DSYSCONFDIR="/usr/local/etc"' -DHAVE_LIBSYSTEMD=1 -MD -MQ xdg-desktop-portal-termfilechooser.p/src_filechooser_filechooser.c.o -MF xdg-desktop-portal-termfilechooser.p/src_filechooser_filechooser.c.o.d -o xdg-desktop-portal-termfilechooser.p/src_filechooser_filechooser.c.o -c ../src/filechooser/filechooser.c
../src/filechooser/filechooser.c: In function ‘exec_filechooser’:
../src/filechooser/filechooser.c:76:15: error: comparison is always true due to limited range of data type [-Werror=type-limits]
   76 |   } while (cr != EOF);
      |               ^~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.

**Comment by sxyazi** [comment]
See https://yazi-rs.github.io/docs/tips#system-chooser
